src/main/java/org/jooq/lambda/concurrent/Async.java

```diff
@@ -0,0 +1,20 @@
+package org.jooq.lambda.concurrent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public final class Async {
+    private Async() {
+
+    }
+
+    public static <U> CompletionStage<U> supplyAsync(Supplier<U> supplier, Executor executor) {
+        return SameExecutorCompletionStage.of(CompletableFuture.supplyAsync(supplier, executor), executor);
+    }
+
+    public static CompletionStage<Void> runAsync(Runnable runnable, Executor executor) {
+        return SameExecutorCompletionStage.of(CompletableFuture.runAsync(runnable, executor), executor);
+    }
```

Comment:
I'm slightly opposed to it just because I think it will encourage users to overuse the common pool.

`Async.runAsync(() -> {...}, ForkJoinPool.commonPool())` isn't that hard to write out if that is what they really want.

I think it's fine to add to match the `CompletableFuture` api but not adding it might encourage better use. Whatever you think.